compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -2263,13 +2263,22 @@ impl<'a, 'tcx> InferCtxtPrivExt<'a, 'tcx> for InferCtxt<'a, 'tcx> {
                                     trait_impls.non_blanket_impls().len()
                                 )
                             };
-
+                            let mut suggestions = vec![(
+                                trait_path_segment.ident.span.shrink_to_lo(),
+                                format!("<{} as ", self.tcx.def_path(impl_def_id).to_string_no_crate_verbose())
```

Comment:
sidenote, using `self.tcx.def_path(impl_def_id).to_string_no_crate_verbose()` here feels wrong, I would expect us to use `self.tcx.type_of(impl_def_id)` here to print the self type. rn this is causing us to print `::StructA` instead of `StructA`.

@lyming2007 would you be interested in changing this in a followup pr?

Reply:
both `<::StructA as TraitA<i32>>::func()` and `<StructA as TraitA<i32>>::func()` will work for the compiler.
Do we really need to change it from `::StructA` to `StructA`?

Reply:
we don't **need** to do it, but it's a lot more idiomatic to use `StructA` instead of `::StructA`. Using `::StructA` is pretty confusing to most people, so it's better to avoid imo

Reply:
see this PR https://github.com/rust-lang/rust/pull/102670 @lcnr